controllers/mutation_reconcile_looper.go

```diff
@@ -91,142 +90,60 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, s
 		return "", fmt.Errorf("resource data cannot be empty")
 	}
 
-	// get config resource
-	config := &configdata.ConfigData{}
-	// TODO: allow for snapshots to be sources here. The chain could be working on an already modified source.
-	resourceRef := spec.ConfigRef.Resource.ResourceRef
-	if resourceRef == nil {
-		return "",
-			fmt.Errorf("resource ref is empty for config ref")
-	}
-	reader, _, err := m.OCMClient.GetResource(ctx, componentVersion, *resourceRef)
-	if err != nil {
-		return "",
-			fmt.Errorf("failed to get resource: %w", err)
-	}
-	defer reader.Close()
-
-	// This content might be Tarred up by OCM.
-	uncompressed, _, err := compression.AutoDecompress(reader)
-	if err != nil {
-		return "", fmt.Errorf("failed to auto decompress: %w", err)
-	}
-	defer uncompressed.Close()
-
-	content, err := io.ReadAll(uncompressed)
-	if err != nil {
-		return "", fmt.Errorf("failed to read blob: %w", err)
-	}
-
-	if err := ocmruntime.DefaultYAMLEncoding.Unmarshal(content, config); err != nil {
-		return "",
-			fmt.Errorf("failed to unmarshal content: %w", err)
-	}
-
-	log.Info("preparing localization substitutions")
+	var (
+		identity  v1alpha1.Identity
+		sourceDir string
+	)
 
-	componentDescriptor, err := component.GetComponentDescriptor(ctx, m.Client, spec.ConfigRef.Resource.ResourceRef.ReferencePath, componentVersion.Status.ComponentDescriptor)
-	if err != nil {
-		return "", fmt.Errorf("failed to get component descriptor from version")
-	}
-	if componentDescriptor == nil {
-		return "", fmt.Errorf("couldn't find component descriptor for reference '%s' or any root components", spec.ConfigRef.Resource.ResourceRef.ReferencePath)
-	}
+	if spec.ConfigRef != nil {
+		var rules localize.Substitutions
```

Comment:
Looks like this is now a single-use variable, so there is no need to declare it here anymore.

Reply:
It needs to be declared here because it's returned from the same function that gives us the identity on L118, and identity is in a different scope.

Reply:
Yep, will do!